app/assets/javascripts/components/generic_object/generic-object-table-component.js

```diff
@@ -0,0 +1,128 @@
+ManageIQ.angular.app.component('genericObjectTableComponent', {
+  bindings: {
+    keys: '=',
+    values: '=',
+    keyType: '@',
+    tableHeaders: '=',
+    valueOptions: '=',
+    newRecord: '=',
+    noOfRows: '=',
+    origKeysValues: '=',
+    tableChanged: '=',
+    requiredRule: '@?',
+    tableRendered: '=',
+    uniqueProperty: '&',
+    angularForm: '=',
+  },
+  controllerAs: 'vm',
+  controller: genericObjectTableController,
+  templateUrl: '/static/generic_object/generic_object_table.html.haml',
+});
+
+genericObjectTableController.$inject = ['$timeout'];
+
+function genericObjectTableController($timeout) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.tableHeaders.push('', '');
+
+    vm.dupicatePropertyError = [];
+    vm.invalidKeyName = __("Invalid name");
+    vm.tableRendered = true;
+
+    if (vm.newRecord) {
+      vm.addRow(0, vm.keyType + '0', true);
+    }
+  };
+
+  vm.addRow = function(_currentRow, element, addFromOtherSource) {
+    vm.keys.push('');
+    vm.noOfRows = _.size(vm.keys);
+
+    if (! addFromOtherSource) {
+      $timeout(function() {
+        angular.element('#' + element).focus();
+      }, -1);
```

Comment:
what does the `-1` do here? You probably don't want a second param at all, right?

Reply:
IIRC, a while back I had read somewhere that `-1` is equivalent to no particular value specified, as in, let the operation take the time it needs before it sets the focus.
...but I did not find that documented anywhere in my search today.

I will remove the `-1` - seems to work OK without it. Thanks.

Reply:
Fixed in c756766